src/ServiceControl/SagaAudit/SagaUpdatedHandler.cs

```diff
@@ -1,24 +1,102 @@
 ﻿namespace ServiceControl.SagaAudit
 {
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
     using System.Threading.Tasks;
     using EndpointPlugin.Messages.SagaState;
+    using Newtonsoft.Json.Linq;
     using NServiceBus;
-    using ServiceControl.Persistence;
+    using ServiceControl.Connection;
+    using ServiceControl.Infrastructure.DomainEvents;
 
     class SagaUpdatedHandler : IHandleMessages<SagaUpdatedMessage>
     {
-        public SagaUpdatedHandler(ISagaAuditDataStore store)
+        public SagaUpdatedHandler(IDomainEvents domainEvents, IPlatformConnectionBuilder connectionBuilder)
         {
-            this.store = store;
+            this.domainEvents = domainEvents;
+            this.connectionBuilder = connectionBuilder;
         }
 
-        public Task Handle(SagaUpdatedMessage message, IMessageHandlerContext context)
+        public async Task Handle(SagaUpdatedMessage message, IMessageHandlerContext context)
         {
-            var sagaSnapshot = SagaSnapshotFactory.Create(message);
+            await WarnOncePerEndpointPerDay(message);
 
-            return store.StoreSnapshot(sagaSnapshot);
+            if (auditQueueName is null || nextAuditQueueNameRefresh < DateTime.UtcNow)
+            {
+                await RefreshAuditQueue();
+            }
+
+            if (auditQueueName is null)
+            {
+                throw new Exception("Could not determine audit queue name to forward saga update message. The ServiceControl remote audit instance ");
+            }
+
+            await context.ForwardCurrentMessageTo(auditQueueName);
+        }
+
+        async Task RefreshAuditQueue()
+        {
+            await semaphore.WaitAsync();
+            try
+            {
+                if (auditQueueName != null && nextAuditQueueNameRefresh > DateTime.UtcNow)
+                {
+                    return;
+                }
+
+                var connectionDetails = await connectionBuilder.BuildPlatformConnection();
+
+                if (connectionDetails.ToDictionary().TryGetValue("SagaAudit", out var sagaAuditObj) && sagaAuditObj is JObject sagaAudit)
+                {
+                    auditQueueName = sagaAudit["SagaAuditQueue"].Value<string>();
+                    nextAuditQueueNameRefresh = DateTime.UtcNow.AddMinutes(5);
```

Comment:
Magic number, I would prefer this to be a readonly TimeSpan where this value is also logged when set

Reply:
I don't know why you want it to be a separate field, especially since I ended up adding a second timeout for failures so that ingestion does not back up, but I did add the logging so you'll know what's going on from the logs.